### PR TITLE
Component: vtgate Auto Update tables in "Status For Vtgate" page

### PR DESCRIPTION
**Pull request by jwangace**
## Description
Currently, for each component that has **/debug/status** page, the page is static and does not refresh containing tables. Vtgate page is particularly useful, as a daily user, I'd like to see those tables to be refreshed, this is quite useful when doing deployments && updates.

This PR, added **refreshTableContentIfTableHasID()** javascript function, it access "/debug/status" in the background(every 5 seconds) and looks for those tables ONLY if they have **id** html attribute defined, and only updates those tables.

Because I only care about the status page of Vtgate, I only added the id html attribute on those relevant tables, but it can be applied to any component which has a static page.

## Related Issue(s)
There is None, this is rather a new feature.

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by GuptaManan100 — COMMENTED** [review]
I really like the idea of auto-refresh. My only comment is to not overload the id field with information. Instead we should create a class (maybe call it `refreshRequired` or something else), and refresh tables that have that class. Using id doesn't seem like a great solution, but it isn't obvious to other developers that adding an id element to a table will cause it to be refreshed.

**Comment by frouioui** [comment]
> I really like the idea of auto-refresh. My only comment is to not overload the id field with information. Instead we should create a class (maybe call it refreshRequired or something else), and refresh tables that have that class. Using id doesn't seem like a great solution, but it isn't obvious to other developers that adding an id element to a table will cause it to be refreshed.

I approve this comment, good idea!

File: go/vt/servenv/status.go [hunk 1]
```diff
@@ -81,6 +81,26 @@ var statusHTML = `<!DOCTYPE html>
...
+  	    if (newTable.length > 0) {
+  	  	  $(this).empty().append(newTable.html());
+  	    }
+  	  }
+    });
+  });
+}
+$(document).ready(function() {
+	setInterval(refreshTableContentIfTableHasID, 5000);
+});
```
Comment (frouioui) [inline go/vt/servenv/status.go:102]: I am not sure whether we should make the interval configurable or not. Not everyone will want to overload their VTGate with a request every 5 seconds.

**Review by frouioui — COMMENTED** [review]
I like the idea of refreshing the page. Thanks for contributing!

**Comment by jwangace** [comment]
> > I really like the idea of auto-refresh. My only comment is to not overload the id field with information. Instead we should create a class (maybe call it refreshRequired or something else), and refresh tables that have that class. Using id doesn't seem like a great solution, but it isn't obvious to other developers that adding an id element to a table will cause it to be refreshed.
> 
> I approve this comment, good idea!

Thank you both @GuptaManan100 and @frouioui for your comments, really helpful. I agree adding class "refreshRequired" is a good idea, it makes things more self-explanatory, definitely can add this class.
However, technically I still think we need to attach an id to each table to make it easier to identify each table, still possible to do that without an id, for example, for this table:
```
<table>
  <tr>
    <th colspan="5">HealthCheck Tablet Cache</th>
  </tr>
  <tr>
   <th>Cell</th>
   <th>Keyspace</th>
   <th>Shard</th>
   <th>TabletType</th>
   <th>TabletHealth</th>
  </tr>
...
</table>
```
The only signature is either the head **HealthCheck Tablet Cache** or the table <th> tags, which is a per table different, this potentially would make javascript code turn ugly pretty quickly.
For now I don't have a good idea but I will put some thoughts on it.

File: go/vt/servenv/status.go (hunk 1 above)
Comment (jwangace) [reply go/vt/servenv/status.go:102]: totally make sense, do you mind giving me suggestions on which place is good for this config and what's the good value as a default?

Comment (jwangace) [reply go/vt/servenv/status.go]: I updated this as a variable, and set it to 10 seconds as default, this should be a value that won't overload vtgate.

**Pull request closed without merging**